framework/src/problems/FEProblemBase.C

```diff
@@ -266,6 +266,14 @@ FEProblemBase::validParams()
       false,
       "Set to true to allow convergence even though the solution has been marked as 'invalid'");
 
+  params.addParam<bool>("immediately_print_invalid_solution",
+                        false,
+                        "Whether or not to report invalid solution warnings at the time the "
+                        "warning is produced instead of after the calculation");
+
+  params.addParamNamesToGroup("allow_invalid_solution immediately_print_invalid_solution",
+                              "SolutionInvalidity Output Control");
```

Comment:
```suggestion
                              "Solution validity control");
```
parameters get added after automatically, see https://mooseframework.inl.gov/source/materials/ParsedMaterial.html